linalg/sparsemat.cpp

```diff
@@ -489,14 +489,15 @@ void SparseMatrix::SortColumnIndices()
                                        d_ja_sorted, sortInfoA,
                                        &pBufferSizeInBytes);
 
-      Array< char > buffer( pBufferSizeInBytes );
-      pBuffer = buffer.Write();
+      CuMemAlloc( &pBuffer, pBufferSizeInBytes );
```

Comment:
What about changing Array so it can allocate larger arrays? Then we could avoid using native cuda/hip calls in the code.

Reply:
I think this would require a lot of changes in MFEM.